### ISSUE TITLE
How to enable php-redis and php-mysql support?

### ISSUE THREAD

**Issue by garretcoffman** [bitnami/bitnami-docker-php-fpm#148]
I'm trying to get php-redis and php-mysql loaded/enabled but having no luck.  Can someone give me the run down on how to add add supports for extensions not already enabled?

**Comment by javsalgar** [comment]
Hi,

The extensions that the container has by default are:

```
[PHP Modules]
bcmath
bz2
calendar
Core
ctype
curl
date
dom
exif
fileinfo
filter
ftp
gd
gettext
gmp
hash
iconv
imap
intl
json
ldap
libxml
mbstring
mysqli
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
pdo_sqlite
Phar
posix
readline
Reflection
session
SimpleXML
soap
sockets
sodium
SPL
sqlite3
standard
tidy
tokenizer
xml
xmlreader
xmlrpc
xmlwriter
xsl
Zend OPcache
zip
zlib

[Zend Modules]
Zend OPcache
```

and

```
root@c09cae5256ce:/app# ls /opt/bitnami/php/lib/php/extensions/ -l
total 3732
-rwxr-xr-x 1 1001 1001   77304 Aug 31 09:44 apcu.so
-rwxr-xr-x 1 1001 1001  474872 Aug 31 09:44 imagick.so
-rwxr-xr-x 1 1001 1001   27416 Aug 31 09:44 maxminddb.so
-rwxr-xr-x 1 1001 1001   56160 Aug 31 09:44 mcrypt.so
-rwxr-xr-x 1 1001 1001  186048 Aug 31 09:44 memcached.so
-rwxr-xr-x 1 1001 1001 1765552 Aug 31 09:44 mongodb.so
-rwxr-xr-x 1 1001 1001  541272 Aug 31 09:44 opcache.so
-rwxr-xr-x 1 1001 1001   31512 Aug 31 09:44 pdo_dblib.so
-rwxr-xr-x 1 1001 1001   44120 Aug 31 09:44 pdo_pgsql.so
-rwxr-xr-x 1 1001 1001  147192 Aug 31 09:44 pgsql.so
-rwxr-xr-x 1 1001 1001  451904 Aug 31 09:44 xdebug.so
```

If you want to add extra extensions, you would need to create a fork of the container where you would add these extensions. Do you know how to build these extensions?

**Comment by garretcoffman** [comment]
No, I'm afraid that's where I get lost.  I just use apt install php-redis in my previous container and that added it but due to the way the bitnami image works doing so in the bitnami container will try to install it in php7.3.

**Comment by javsalgar** [comment]
One thing you could try is to use a multi-stage container. In one you execute the apt install php-redis and in the second you simply copy the .so file.

See examples of multi-stage containers here https://www.docker.com/blog/advanced-dockerfiles-faster-builds-and-smaller-images-using-buildkit-and-multistage-builds/

**Comment by garretcoffman** [comment]
I actually tried that and it wouldn't recognize the file.  I pulled the bitnami container (non-prod version), then used a Dockerfile to run apt install php-fpm within the container, which loaded the file into the php7.3 folder (which made no sense to me), then I copied from the php7.3 folder to the extensions folder in the bitnami phpfpm folder and modified the php.ini file to enable it.  It kept giving me errors that it couldn't load the file.

In the end I just built the whole container from the ubuntu 20.04 image.  It's larger but it works.
